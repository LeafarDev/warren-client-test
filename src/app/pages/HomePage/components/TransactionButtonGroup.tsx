import { Grid } from '@material-ui/core';
import { ButtonTransaction, ButtonTransactionList, GridButtonTransaction } from '../styles';
import React from 'react';

interface Props {
  openTransactionModal(transactionType: string): void;
}

export const TransactionButtonGroup: React.FC<Props> = props => {
  const {
    openTransactionModal
  } = props;

  return (
    <Grid item xs={12} sm={12} md={6}>
      <ButtonTransactionList container direction={'row'} alignItems={'flex-end'}>
        <GridButtonTransaction container item xs={12} sm={4} md={4} lg={3}>
          <ButtonTransaction variant="contained" onClick={() => openTransactionModal('deposit')}>
            New Deposit
          </ButtonTransaction>
        </GridButtonTransaction>
        <GridButtonTransaction item xs={12} sm={4} md={4} lg={3}>
          <ButtonTransaction variant="contained" onClick={() => openTransactionModal('withdraw')}>
            New Withdraw
          </ButtonTransaction>
        </GridButtonTransaction>
        <GridButtonTransaction item xs={12} sm={4} md={4} lg={3}>
          <ButtonTransaction variant="contained" onClick={() => openTransactionModal('payment')}>
            New Payment
          </ButtonTransaction>
        </GridButtonTransaction>
      </ButtonTransactionList>
    </Grid>
  );
};


