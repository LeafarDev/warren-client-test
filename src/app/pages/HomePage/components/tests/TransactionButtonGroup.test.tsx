import React from 'react';
import { TransactionButtonGroup } from '../TransactionButtonGroup';
import { render } from '@testing-library/react';

describe('<TransactionButtonGroup />', () => {
  const openTransactionModal = jest.fn();

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(<TransactionButtonGroup openTransactionModal={openTransactionModal}/>);

    expect(firstChild)
      .toMatchSnapshot();
  });
});
