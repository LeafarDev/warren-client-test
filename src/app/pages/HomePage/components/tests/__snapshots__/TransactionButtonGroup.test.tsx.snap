// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionButtonGroup /> should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
>
  <div
    class="MuiGrid-root sc-dlnjwi hiknWI MuiGrid-container MuiGrid-align-items-xs-flex-end"
  >
    <div
      class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          New Deposit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          New Withdraw
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          New Payment
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
